aws/aws-service-bundler/src/main/java/software/amazon/smithy/java/aws/servicebundler/AwsServiceBundler.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.aws.servicebundler;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import software.amazon.smithy.aws.traits.auth.SigV4Trait;
+import software.amazon.smithy.awsmcp.model.AwsServiceMetadata;
+import software.amazon.smithy.awsmcp.model.Model;
+import software.amazon.smithy.java.core.serde.document.Document;
+import software.amazon.smithy.model.loader.ModelAssembler;
+import software.amazon.smithy.model.node.BooleanNode;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.ModelSerializer;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.EndpointTrait;
+import software.amazon.smithy.modelbundle.api.Bundler;
+import software.amazon.smithy.modelbundle.api.model.Bundle;
+
+final class AwsServiceBundler implements Bundler {
+    private static final ShapeId ENDPOINT_TESTS = ShapeId.from("smithy.rules#endpointTests");
+
+    // visible for testing
+    static final Map<String, String> GH_URIS_BY_SERVICE = new HashMap<>();
+    static {
+        // line is in the form fooService/service/version/fooService.json
+        try (var models = new BufferedReader(new InputStreamReader(
+                Objects.requireNonNull(AwsServiceBundler.class.getResourceAsStream("/models.txt")),
+                StandardCharsets.UTF_8))) {
+            models.lines()
+                    .forEach(line -> GH_URIS_BY_SERVICE
+                            .put(line.substring(0, line.indexOf("/")).toLowerCase(Locale.ROOT), line));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final ModelResolver resolver;
+    private final String serviceName;
+
+    AwsServiceBundler(String... args) {
```

Comment:
Why are we taking a varargs?

Reply:
As opposed to a `String[]`?

Reply:
The constructor you are delegating to takes a single service name and you are just passing the first arg. Maybe I am missing something, but I don't understand why take multiple strings at all, since this class works only on a single service na,e.

Reply:
Because there are other providers that will take more arguments. I'm just passing the CLI args through with the first two arguments – output location and bundler type – redacted. Each bundle implementation is effectively a subcommand of the bundler CLI.